protocols/rendezvous/src/codec.rs

```diff
@@ -236,6 +243,75 @@ impl Decoder for RendezvousCodec {
     }
 }
 
+#[derive(Clone, Default)]
+pub struct Codec {}
+
+#[async_trait]
+impl libp2p_request_response::Codec for Codec {
+    type Protocol = StreamProtocol;
+    type Request = Message;
+    type Response = Message;
+
+    async fn read_request<T>(&mut self, _: &Self::Protocol, io: &mut T) -> io::Result<Self::Request>
+    where
+        T: AsyncRead + Unpin + Send,
+    {
+        if let Some(result) = FramedRead::new(io, RendezvousCodec::default()).next().await {
+            return Ok(result?);
+        }
+
+        Err(io::ErrorKind::InvalidInput.into())
```

Comment:
Any reason you didn't end up going for `ok_or` here?

Reply:
```
let message = framed.next().await.ok_or(io::ErrorKind::InvalidInput)?
```
^ the trait `From<std::result::Result<_, _>>` is not implemented for `std::io::Error`